From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Paul Praschl <praschlpaul@g-p.at>
Date: Sun, 26 Jul 2026 17:17:28 +0200
Subject: [PATCH 037/117] dt-bindings: gpio: add Apple D2365 PMIC GPIOs

Document the 24-line GPIO controller exposed by the D2365 PMIC regmap
and allow it as a child of the Apple PMIC node. Board descriptions can
reserve unavailable lines and use the remaining GPIOs for peripheral
power enables.

Signed-off-by: Paul Praschl <praschlpaul@g-p.at>

Orchard-replay-of: c35e4a8dd20aed76757c6822c8452f853c0fcbde
---
 .../bindings/gpio/apple,d2365-pmic-gpio.yaml  | 48 +++++++++++++++++++
 .../devicetree/bindings/mfd/apple,pmic.yaml   |  3 ++
 2 files changed, 51 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/gpio/apple,d2365-pmic-gpio.yaml

diff --git a/Documentation/devicetree/bindings/gpio/apple,d2365-pmic-gpio.yaml b/Documentation/devicetree/bindings/gpio/apple,d2365-pmic-gpio.yaml
new file mode 100644
index 000000000..d45829d0f
--- /dev/null
+++ b/Documentation/devicetree/bindings/gpio/apple,d2365-pmic-gpio.yaml
@@ -0,0 +1,48 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/gpio/apple,d2365-pmic-gpio.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Apple D2365 PMIC GPIO controller
+
+maintainers:
+  - Paul Praschl <praschlpaul@g-p.at>
+
+description: |
+  The Dialog D2365 PMIC used in Apple A10 devices provides 24 GPIOs. The
+  controller is accessed through the regmap exported by the parent Apple I2C
+  PMIC device.
+
+properties:
+  compatible:
+    const: apple,d2365-pmic-gpio
+
+  gpio-controller: true
+
+  "#gpio-cells":
+    const: 2
+
+  gpio-reserved-ranges:
+    maxItems: 2
+
+required:
+  - compatible
+  - gpio-controller
+  - "#gpio-cells"
+
+additionalProperties: false
+
+examples:
+  - |
+    pmic {
+        #address-cells = <1>;
+        #size-cells = <1>;
+
+        gpio {
+            compatible = "apple,d2365-pmic-gpio";
+            #gpio-cells = <2>;
+            gpio-controller;
+        };
+    };
+...
diff --git a/Documentation/devicetree/bindings/mfd/apple,pmic.yaml b/Documentation/devicetree/bindings/mfd/apple,pmic.yaml
index 22a39a0ac..25b522995 100644
--- a/Documentation/devicetree/bindings/mfd/apple,pmic.yaml
+++ b/Documentation/devicetree/bindings/mfd/apple,pmic.yaml
@@ -55,6 +55,9 @@ properties:
   "#size-cells":
     const: 1
 
+  gpio:
+    $ref: /schemas/gpio/apple,d2365-pmic-gpio.yaml
+
 patternProperties:
   "^charger@[0-9a-f]+$":
     $ref: /schemas/power/supply/apple,d2365-charger.yaml
